javascript/javascript3/week1/weatherApp.js

```diff
@@ -0,0 +1,171 @@
+
+        fetch(`https://api.openweathermap.org/data/2.5/weather?q=${cityName}&appid=1e55d7a6d0d20e41c51f61a54d2abb7d`)
+            .then((res) => res.json())
+            .then((data) => {
+                if (data.cod == 200) {
+                    informationwithCity(data)
+                }
+                else {
+                    errorText()
+                }
+            })
+    }
+    displayTheDatawithCityWithAutomatic()
+}
+
+
+// For errors
+function errorText() {
+    city.innerHTML = 'Please write a correct city!!!'
+}
+
+// With city and with location button events
+
+buttonWithCity.addEventListener('click', displayTheDatawithCity)
+
+locationButton.addEventListener('click', getLatitudeLongitude)
+
+// Get result with city name function
+
+function displayTheDatawithCity() {
+    const cityName = input.value.toLowerCase()
+
+    fetch(`https://api.openweathermap.org/data/2.5/weather?q=${cityName}&appid=1e55d7a6d0d20e41c51f61a54d2abb7d`)
+        .then((res) => res.json())
+        .then((data) => {
+            if (data.cod == 200) {
+                informationwithCity(data)
+            }
+            else {
+                errorText()
+            }
+        })
+}
+
+// Html informations part with city name event
+
+function informationwithCity(data) {
+    city.innerHTML = `City : ${data.name}`
+
+    temprature.innerHTML = `${Math.floor(data.main.temp - 273.15)} °C`
+
+    feelsLike.innerHTML = `<b>Feels : ${Math.floor(data.main.feels_like - 273.15)} °C </b>`
+
+    minMaxTemp.innerHTML = `<b>Max Temp : ${Math.floor(data.main.temp_max - 273.15)} °C - Min Temp: ${Math.floor(data.main.temp_min - 273.15)} °C </b>`
+
+    pressure.innerHTML = `<b>Pressure : ${data.main.pressure} hPa - Humidity: ${data.main.humidity}% </b>`
+
+    img.src = `http://openweathermap.org/img/w/${data.weather[0].icon}.png`
+
+    wind.innerHTML = `Wind Speed : ${data.wind.speed} m/s - Wind Degree : ${data.wind.deg}`
+
+    cloud.innerHTML = `Clouds : ${data.clouds.all} %`
+
+    sunTimes.innerHTML = `Sunrise Time: ${sun(data.sys.sunrise)} - Sunset Time : ${sun(data.sys.sunset)}`
+
+    mapLocation.style.visibility = 'visible'
+    const srcLoc = `https://maps.google.com/maps?q=${data.coord.lat},${data.coord.lon}&z=15&output=embed`
+    mapLocation.src = srcLoc
+}
+
+// Get results with automatically latitude and longitude
+
+function getLatitudeLongitude() {
+    function confirmed(location) {
+        const latitude = location.coords.latitude
+        const longitude = location.coords.longitude
+        fetch(`http://api.openweathermap.org/data/2.5/onecall?lat=${latitude}&lon=${longitude}&appid=1e55d7a6d0d20e41c51f61a54d2abb7d`)
+            .then(res => res.json())
+            .then((data) => {
+                if (data.cod != 400) {
+                    informationwithLocation(data)
+                }
+                else {
+                    error()
+                    console.log(data)
+                }
+            })
```

Comment:
You use almost exactly the same lines of code at the 3 places you do API calls in this file. You could make it into a single function that takes a URL and a function to call if the API call is succesful (in this case you use informationWithLocation). 
It would make your life much easier when you have to change something in the code, because you only have to change it in one place. It would also let you only define your API key in one place instead of 3 different places 😃